### PR TITLE
Update megatron interface to dialogue 

### PR DESCRIPTION
**Pull request by Zhilin123**
# What does this PR do ?

Update megatron interface to dialogue 

**Collection**: NLP

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Review by vadam5 — APPROVED** [review]
Overall looks good, just left two minor comments

**Review by vadam5 — CHANGES_REQUESTED** [review]
Just a couple small things, please see comment below

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py [hunk 1]
```diff
@@ -397,12 +397,10 @@ def forward(
                 input_embeds = self.embed_input_inference(input_ids, taskname_ids)
             else:
                 input_embeds = self.embed_input_train(input_ids, taskname_ids)
-
             position_embeddings = self.frozen_model.model.language_model.embedding.position_embeddings(position_ids)
             encoder_input = input_embeds + position_embeddings
         else:
             encoder_input = None
-
```
[the remark this answers is not in the document]
Comment (Zhilin123) [reply nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py:405]: sure

File: examples/nlp/dialogue/conf/dialogue_config.yaml [hunk 2]
```diff
@@ -65,6 +60,28 @@ model:
     num_output_layers: 2
     fc_dropout: 0.1
   
+  # Dialogue GPT Classification Magetron Prompt Learning model args
+  prompt_learning: false # please change to true to activate prompt learning
+  language_model_path: null
```
[the remark this answers is not in the document]
Comment (Zhilin123) [reply examples/nlp/dialogue/conf/dialogue_config.yaml:65]: sure

**Pull request closed without merging**